### PR TITLE
Public WorkflowLayout, tidier WorkflowRunner, bye WorkflowActivityRunner

### PR DESCRIPTION
**Pull request by rjrjr**

### CODE REVIEW

File: kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunner.kt [hunk 1]
```diff
@@ -28,4 +44,161 @@ interface WorkflowRunner<out OutputT> {
...
+}
+
+/**
+ * If your workflow needs to manage the back button, override [FragmentActivity.onBackPressed]
+ * and call this method, and have your views or coordinators use [HandlesBack].
+ *
+ * e.g.:
+ *
+ *    override fun onBackPressed() {
+ *      if (!workflowViewModel.onBackPressed(this)) super.onBackPressed()
```
Comment (zach-klippenstein) [inline kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunner.kt]: Should this variable be named `runner` to be consistent with the previous code snippets?

Comment (rjrjr) [reply kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunner.kt]: Yes, thanks, that's stale.

**Comment by rjrjr** [comment]
Rebased to target branch v0.16-fixes. Updated with convenience overloads of `WorkflowLayout.of`.

File: kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowFragment.kt [hunk 2]
```diff
@@ -63,25 +62,38 @@ import io.reactivex.Observable
 @ExperimentalWorkflowUi
 abstract class WorkflowFragment<InputT, OutputT : Any> : Fragment() {
 
-  data class Config<InputT, OutputT : Any>(
+  data class Configured<InputT, OutputT : Any> internal constructor(
```
Comment (zach-klippenstein) [inline kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowFragment.kt]: Nit: past tense seems like an odd choice for this name, I kinda liked `Config` better but i don't care much at all.

File: kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunner.kt [hunk 3]
```diff
@@ -28,4 +48,237 @@ interface WorkflowRunner<out OutputT> {
...
+ */
+@ExperimentalWorkflowUi
+@CheckResult
+fun <InputT, OutputT : Any, RenderingT : Any> FragmentActivity.setContentWorkflow(
+  viewRegistry: ViewRegistry,
+  workflow: Workflow<InputT, OutputT, RenderingT>,
+  input: InputT,
+  savedInstanceState: Bundle?
+): WorkflowRunner<OutputT> {
+  return setContentWorkflow(viewRegistry, workflow, Observable.just(input), savedInstanceState)
```
Comment (zach-klippenstein) [inline kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunner.kt]: Nit: `Flowable.just`

File: kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowFragment.kt (hunk 2 above)
Comment (rjrjr) [reply kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowFragment.kt]: Agreed.

**Pull request merged**